src/analysis/vector/qgsgeometrysnapper.cpp

```diff
@@ -102,7 +102,7 @@ bool QgsSnapIndex::SegmentSnapItem::getProjection( const QgsPoint &p, QgsPoint &
 bool QgsSnapIndex::SegmentSnapItem::withinDistance( const QgsPoint &p, const double tolerance )
 {
   double minDistX, minDistY;
-  const double distance = QgsGeometryUtilsBase::sqrDistToLine( p.x(), p.y(), idxFrom->point().x(), idxFrom->point().y(), idxTo->point().x(), idxTo->point().y(), minDistX, minDistY, 4 * std::numeric_limits<double>::epsilon() );
+  const double distance = std::sqrt( QgsGeometryUtilsBase::sqrDistToLine( p.x(), p.y(), idxFrom->point().x(), idxFrom->point().y(), idxTo->point().x(), idxTo->point().y(), minDistX, minDistY, 4 * std::numeric_limits<double>::epsilon() ) );
```

Comment:
If you want to use the distance, use the dedicated method:

```suggestion
  const double distance = QgsGeometryUtilsBase::distance2D( p.x(), p.y(), idxFrom->point().x(), idxFrom->point().y(), idxTo->point().x(), idxTo->point().y(), minDistX, minDistY, 4 * std::numeric_limits<double>::epsilon() );
```
I also seem to have seen this in the geometry checker, maybe; I thought it was deliberate (to avoid using square root) and to compare double the distance; this doesn't seem to be the case in processing. Have you checked to be certain?

Reply:
At the very least if this was deliberate, ``tolerance`` should be renamed to ``toleranceSquared``

Reply:
I would also prefer to compare the distance squared to the tolerance squared rater then adding a root square calculation.

Reply:
I think that was an oversight in https://github.com/qgis/QGIS/commit/c8179c36c59f3d8058d80efeaab3c2ff02d92118.

Reply:
@agiudiceandrea , adjusted.